README.md

```diff
@@ -99,101 +140,122 @@ var excitedWords = excite(wordList);
 
-In this example, we have a function that toggles the opacity of an image (when the opacity is 0, it cannot be seen). Once the function is called, if the image is visible, it becomes invisible, but if it is already invisible, it instead becomes visible.
+Most programs we want to write wouldn't work if we completely disallowed side
+effects. How then can we ensure that our impure functions are testable?
 
-In the first version below, the function takes no arguments, alters the DOM after checking the global variable `changeTransition`, then changes the the global variable `changeTransition`.
+In this example, we have a function that toggles the opacity of an image (when
+the opacity is 0, it cannot be seen). Once the function is called, if the image
+is visible, it becomes invisible, but if it is already invisible, it instead
+becomes visible.
+
+In the first version below, the function takes no arguments, alters the DOM
+after checking the global variable `changeTransition`, then changes the the
+global variable `changeTransition`.
 
 ```js
 var changeTransition = true;
 
-function visionChange () {
-
-  var visionimage = document.getElementById('visionimage');
+function visionChange() {
+  var visionimage = document.getElementById("visionimage");
 
   if (changeTransition === true) {
-  visionimage.style.opacity = 0;
+    visionimage.style.opacity = 0;
   }
 
   if (changeTransition === false) {
     visionimage.style.opacity = 1;
   }
 
-
   if (changeTransition === true) {
-      changeTransition = false;
+    changeTransition = false;
   } else {
-      changeTransition = true;
+    changeTransition = true;
   }
-
 }
 ```
-Next we rewrite the function as two separate functions, both of which will return the same value every time, when given the same argument.
 
-Can you see what each functions return? How might you test the functions?
+Next we rewrite the function as three separate functions, both of them will
+return the same value every time, when given the same argument.
 
+Can you see what each functions return? How might you test the functions?
 
 ```js
-function visionChange (changeTransition) {
-  if (changeTransition){
-      return false;
+function transitionToggle(changeTransition) {
+  if (changeTransition) {
+    return false;
   } else {
-      return true;
+    return true;
   }
 }
 
-function updateDom (changeTransition) {
-  return function() {
-    var visionimage = document.getElementById('visionimage');
-
-    if (changeTransition) {
-        visionimage.style.opacity = 0;
-    } else {
-        visionimage.style.opacity = 1;
-    }
+function opacityPicker(changeTransition) {
+  if (changeTransition) {
+    return 0;
+  } else {
+    return 1;
   }
 }
 
-var impureUpdateDom = updateDom(visionChange(changeTransition));
+function updateDom(newOpacity) {
+  var visionimage = document.getElementById("visionimage");
+  visionimage.style.opacity = newOpacity;
+}
+
+var impureUpdateDom = updateDom(
```

Comment:
This is kind of what I imagined, but there's a couple of things we can still simplify; we shouldn't artificially complicate the example.

Ultimately, what we want to demonstrate in this example is the separation of the impure code from the pure code. In this case: the code that manipulates the DOM from the code that determines how the DOM should be updated.

I think your changes to the `updateDom` function are fine. But I think it makes more sense to combine `opacityPicker` and `transitionToggle` together; I understand the idea behind separating the calculation of state (`transitionToggle`) from the calculation of actual style values (`opacityPicker`), but I think in such a simple example, it's not clear why such a separation is being made.

You may wish to consider an alternative to `updateDom`: an `updateOpacity` function which accepts a `HTMLElement` and an opacity value:
```js
function updateOpacity (element, opacity) {
  element.styles.opacity = opacity;
}
```
This offers the benefit of slightly better testability. For example:
```js
test('updateOpacity', function (t) {
  var element = { styles: { opacity: 0 } };
  updateOpacity(element, 1);
  t.equal(element.styles.opacity, 1);
  t.end();
});
```

whereas in the actual code, you could call 
```js
updateOpacity(document.getElementById('blah'), opacityPicker(changeTransition))
```